lib/pbench/test/functional/server/test_put.py

```diff
@@ -55,13 +54,19 @@ def test_index_all(self, server_client: PbenchServerClient, login_user):
                     dataset.resource_id, ["dataset.state", "server.status"]
                 )
                 state = metadata["dataset.state"]
-                assert state in ["Uploaded", "Indexing", "Indexed"]
-                if state == "Indexed":
-                    assert metadata["server.status"]["backup"] == "ok"
-                    assert metadata["server.status"]["unpack"] == "ok"
-                    assert metadata["server.status"]["index"] == "ok"
+                status = metadata["server.status"]
+                stats = set(status.keys()) if status else set()
+                if state == "Indexed" and {"unpack", "index"} <= stats:
```

Comment:
If the `state` is `"Indexed"`, is it possible that `"index"` would be missing from `stats`?  (Is there a race condition there?)

Should this check be omitting the `state` conditional?

Reply:
Since we're not using macro-transactions, yeah, in theory we could fail if we hit at just the right fraction of a second. Integrating `Dataset.advance` with `Sync.update` would help (especially if I can figure out how to do it under a manual transaction without breaking unit tests or production). We've had that discussion under #3080 and I'll give the same response here: better atomicity would help a lot of things, but it's not easy and I'd rather get this fix out without trying to boil the cosmos.